src/descriptor/mod.rs

```diff
@@ -420,6 +676,17 @@ impl<Pk: MiniscriptKey + ToPublicKey> Descriptor<Pk> {
     }
 }
 
+impl Descriptor<DescriptorPublicKey> {
+    /// Derives all wildcard keys in the descriptor using the supplied `path`
+    pub fn derive(&self, path: &[bip32::ChildNumber]) -> Descriptor<DescriptorPublicKey> {
+        self.translate_pk(
+            |pk| Result::Ok::<DescriptorPublicKey, ()>(pk.clone().derive(path)),
+            |pkh| Result::Ok::<hash160::Hash, ()>(*pkh),
+        )
```

Comment:
Actually *this* is wrong: it won't translate hashes for the new public key. This prevents the implementation of `lookup_pkh_sig` for the satisfier.

Reply:
The only way around i see is to store the actual public key as well as the pkh in the enum variant.

Reply:
I don't see how this breaks anything. Keys that are referred to by hash just won't be derived (because we can't), just as with non-wildcard xpubs and normal pubs. Can you point me to the part in the satisfier code where you think this derive impl breaks invariants please? Maybe I'm just missing something.

Reply:
Actually i came across this using it on my end.

I have those scripts for revault in which the policy compiler is optimizing part of the pubkeys to pubkey hashes.
I use a `descriptor_with_master_xpubs.derive([some, path])` kind of logic, which does not derive the hashes.

I could re-generate a descriptor each time i want to use a new depth, but that's... killing the purpose of having a derive() method.
I think that deriving only part of the keys is also an unexpected api: we could instead store the `Pk` as well as the `Pkh` when translating pubkeys to pubkey hashes.

Reply:
Yeah, I mean even the `p2pkh` descriptor uses the key type, there's now way to construct this directly from a hash, so why not also keep the key in miniscript? The conversion to a hash can happen when needed. It should be clear from the context (`pkh`) anyway.